### DISCUSSION TITLE
Create Private Repo for Distro Upgrades

### DISCUSSION THREAD

**Discussion by ferrellsl** [Joshua-Riek/ubuntu-rockchip#64] [General]
I'm a novice Linux user but I do a lot of programming and testing within Linux.  It's quite a tedious process to reinstall all my apps, tools  and settings every time there's a distro upgrade for my Pi.  Would it be possible to implement a private repo such as the ones below so that we can perform a distribution upgrade instead of having to do a complete re-install/re-flash on every point release?

https://linuxopsys.com/topics/create-your-own-repository-for-packages-on-debian

https://linuxconfig.org/how-to-create-a-ubuntu-repository-server

**Comment by Joshua-Riek** [comment]
I have things primarily stable with the OS now and will start working on an upgrade process. I have some ideas on how to implement that already, I like how [this](https://github.com/TheRemote/Ubuntu-Server-raspi4-unofficial) repo set up the upgrade process for a pi4 distro. 

It will take some time though, a fair amount of changes between updates can't be done with a simple apt update/install.

**Comment by Joshua-Riek** [comment]
But creating a script to update the firmware, kernel, bootloader, and GPU-specific packages should not be too hard.

**Reply by ferrellsl** [reply]
That would be perfect!
